### ISSUE TITLE
CrossEntropyLoss的在label中存在-100时，loss的计算不准确

### ISSUE THREAD

**Issue by wn1652400018** [PaddlePaddle/Paddle#47418]
### bug描述 Describe the Bug

paddle版本2.3.2
我在比较torch和paddle的CrossEntropyLoss时发现，如果label中存在-100，那么paddle和torch的计算结果不一样（如果没有-100则loss相同）。
我猜测大概时paddle在统计参与平均只计算的总个数时把label为-100的也考虑进去了。可以通过实验来验证代码如下：
`import torch
import paddle
import numpy as np
label = np.random.randint(0, 7, (511,)).astype(np.int64)
logit = np.random.randn(511, 7)
label[0:100] = -100
label[-100:] = -100
ct = torch.nn.CrossEntropyLoss()
cp = paddle.nn.CrossEntropyLoss()
l1, l2, l3 = ct(torch.tensor(logit), torch.tensor(label)), cp(paddle.to_tensor(logit[100:-100]), paddle.to_tensor(label[100:-100])),\
cp(paddle.to_tensor(logit), paddle.to_tensor(label))

l1, l2, l3 = l1.numpy(), l2.numpy(), l3.numpy()

print(np.abs(l1 - l2) < 1e-6) # 这里验证了个人猜测
print(np.abs(l1 - l3 * 511 / (511 - 200)) < 1e-6)`

**Comment by huangjun12** [comment]
是的，label中存在-100时，计算平均时会将其考虑进去，可以按你描述的方法进行修正
